xarray/core/dataset.py

```diff
@@ -890,11 +887,11 @@ def _replace(  # type: ignore
                 self._coord_names = coord_names
             if dims is not None:
                 self._dims = dims
-            if attrs is not self.__default:
+            if not isinstance(attrs, Default):
```

Comment:
I feel like there's some Chesterton's fence here: is there some reason we had `self.` here?

Reply:
I don't think so. ``self.__default`` is misleading - it was a class attribute to begin with (you could rewrite it as ``Dataset.__default``).